modules/contentpool_remote_register/src/Plugin/rest/resource/ContentpoolTermReferenceFieldsResource.php

```diff
@@ -0,0 +1,157 @@
+use Drupal\rest\ModifiedResourceResponse;
+use Drupal\rest\Plugin\ResourceBase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * A rest resource that provides term reference fields for satellites.
+ *
+ * @RestResource(
+ *   id = "contentpool:contentpool_term_reference_fields",
+ *   label = "Contentpool term reference fields",
+ *   uri_paths = {
+ *     "canonical" = "/api/contentpool-term-reference-fields",
+ *   }
+ * )
+ */
+class ContentpoolTermReferenceFieldsResource extends ResourceBase {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * RemoteRegistrationResource constructor.
+   *
+   * @param array $configuration
+   *   The configuration array.
+   * @param string $plugin_id
+   *   The plugin id.
+   * @param mixed $plugin_definition
+   *   The plugin definition.
+   * @param array $serializer_formats
+   *   The array of serializer formats.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The logger service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager
+   *   The entity field manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, array $serializer_formats, LoggerInterface $logger, EntityTypeManagerInterface $entity_type_manager, EntityFieldManagerInterface $entity_field_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $serializer_formats, $logger);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entity_field_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->getParameter('serializer.formats'),
+      $container->get('logger.factory')->get('rest'),
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * Provides a response for the endpoint.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request object.
+   *
+   * @return \Drupal\rest\ModifiedResourceResponse
+   *   Response.
+   */
+  public function get(Request $request) {
+    $entity_type = $request->query->get('entity_type');
+    $bundle = $request->query->get('bundle');
+
+    $response_data = [];
+
+    $field_definitions = $this->entityFieldManager->getFieldDefinitions($entity_type, $bundle);
```

Comment:
seems like this errors when query parameters are missing?

Reply:
no, it will just return an empty array